presto-main/src/main/java/com/facebook/presto/execution/buffer/ArbitraryOutputBuffer.java

```diff
@@ -237,10 +250,12 @@ public void enqueue(List<SerializedPage> pages)
     }
 
     @Override
-    public void enqueue(int partition, List<SerializedPage> pages)
+    public void enqueue(Lifespan lifespan, int partition, List<SerializedPage> pages)
     {
+        requireNonNull(lifespan, "lifespan is null");
```

Comment:
This is check in the overriden `enqueue` method. I think this check is extra here.

Reply:
I think the same argument applies to `lifespan` argument